### PR TITLE
[FLINK-11635][docs-zh] translate dev/stream/state/checkpointing into Chinese

### PR DESCRIPTION
**Pull request by Henvealf**
Translate the checkpointing doc into Chinese。

### CODE REVIEW

**Comment by wuchong** [comment]
Hi @Henvealf , this is not a pull request for FLINK-14227. Please correct it in the title. Thanks.

**Comment by yangjf2019** [comment]
Hi @Henvealf ,Thanks for your effort. [here](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications) are some Flink Translation Specifications .Maybe it is useful to you.

File: docs/dev/stream/state/checkpointing.zh.md [hunk 1]
```diff
@@ -25,146 +25,138 @@ under the License.
...
-Every function and operator in Flink can be **stateful** (see [working with state](state.html) for details).
-Stateful functions store data across the processing of individual elements/events, making state a critical building block for
-any type of more elaborate operation.
+Flink 中的每个方法或算子都能够被**状态化**（阅读 [working with state](state.html) 查看详细）。
+状态化的方法在处理单个 元素/事件 的时候存储数据，让状态成为使各个类型的算子更加精细的重要部分。
+为了让状态容错，Flink 需要为状态添加**检查点（Checkpoint）**。检查点允许 Flink 恢复状态，并且能够在流中定位，让应用像无故障一样的运行。
 
-In order to make state fault tolerant, Flink needs to **checkpoint** the state. Checkpoints allow Flink to recover state and positions
-in the streams to give the application the same semantics as a failure-free execution.
+[Documentation on streaming fault tolerance]({{ site.baseurl }}/internals/stream_checkpointing.html) 介绍了 Flink 流计算容错机制的内部技术原理。
```
Comment (KarmaGYZ) [inline docs/dev/stream/state/checkpointing.zh.md]: Shoule be `{{ site.baseurl }}/zh/internals`

File: docs/dev/stream/state/checkpointing.zh.md [hunk 2]
```diff
@@ -25,146 +25,138 @@ under the License.
...
+    注意这个值也意味着并发检查点的数目是*一*。
 
-  - *prefer checkpoint for recovery*: This determines if a job will fallback to latest checkpoint even when there are more recent savepoints available to potentially reduce recovery time.
+  - *并发检查点的数目*: 默认情况下，系统不会在有一个检查点在进行时触发另一个检查点。这确保了拓扑不会再检查点上花费太多时间，并且不会和流处理一块进行。
+    在感兴趣的 pipelines 上有确定的处理延迟（例如是因为某方法所调用的外部服务需要些时间来回复），但是仍然想进行频繁的检查点去最小化故障重跑时，允许多个检查点重叠是可能的。
+    
+    该选项在检查点间的最小时间被定义时不能使用。
+    
+  - *外部检查点（externalized checkpoints）*: 你可以配置周期检查点存储到外部系统中。外部检查点将他们的元数据写到持久化存储上并且在 job 失败的时候*不会*被自动删除。
+    这种方式下，如果你的 job 失败，你将会有一个现有的检查点去恢复。更多的细节请看 [deployment notes on externalized checkpoints]({{ site.baseurl }}/ops/state/checkpoints.html#externalized-checkpoints)。
```
Comment (KarmaGYZ) [inline docs/dev/stream/state/checkpointing.zh.md]: Same as above `{{ site.baseurl }}/zh/ops`

File: docs/dev/stream/state/checkpointing.zh.md [hunk 3]
```diff
@@ -25,146 +25,138 @@ under the License.
...
 env.get_checkpoint_config().set_prefer_checkpoint_for_recovery(True)
 {% endhighlight %}
 </div>
 </div>
 
-### Related Config Options
+### 相关的配置选项
 
-Some more parameters and/or defaults may be set via `conf/flink-conf.yaml` (see [configuration]({{ site.baseurl }}/ops/config.html) for a full guide):
+更多的属性 与/或 默认值能在 `conf/flink-conf.yaml` 中设置（完整教程请阅读 [configuration]({{ site.baseurl }}/ops/config.html)）。
```
Comment (KarmaGYZ) [inline docs/dev/stream/state/checkpointing.zh.md]: Same as above.

**Comment by Henvealf** [comment]
> Hi @Henvealf , this is not a pull request for FLINK-14227. Please correct it in the title. Thanks.

Which title should I use, FLINK-14313 ?

**Comment by wuchong** [comment]
I think it should be FLINK-11635. I updated the title.

Hi @xintongsong , do you have other concerns on this?

**Comment by xintongsong** [comment]
Thanks @wuchong, I do have other concerns.

I've left several comments, regarding translations that IMO are not easy to understand. These comments are marked 'resolved' by @Henvealf without addressed or any further explanation.

**Comment by Henvealf** [comment]
> Thanks @wuchong, I do have other concerns.
> 
> I've left several comments, regarding translations that IMO are not easy to understand. These comments are marked 'resolved' by @Henvealf without addressed or any further explanation.

I have commit the update.

File: docs/dev/stream/state/checkpointing.zh.md [hunk 4]
```diff
@@ -25,146 +25,138 @@ under the License.
...
-// allow only one checkpoint to be in progress at the same time
+// 同一时间只允许一个 checkpoint 进行
 env.getCheckpointConfig().setMaxConcurrentCheckpoints(1);
 
-// enable externalized checkpoints which are retained after job cancellation
+// 激活在 job 中止后仍然保留的外部 checkpoint
 env.getCheckpointConfig().enableExternalizedCheckpoints(ExternalizedCheckpointCleanup.RETAIN_ON_CANCELLATION);
 
-// allow job recovery fallback to checkpoint when there is a more recent savepoint
+// 允许在有更近保存点时回退到 checkpoint
```
Comment (xintongsong) [inline docs/dev/stream/state/checkpointing.zh.md]: savepoint不翻译

File: docs/dev/stream/state/checkpointing.zh.md [hunk 5]
```diff
@@ -25,146 +25,138 @@ under the License.
...
-# allow only one checkpoint to be in progress at the same time
+# 同一时间只允许一个 checkpoint 进行
 env.get_checkpoint_config().set_max_concurrent_checkpoints(1)
 
-# enable externalized checkpoints which are retained after job cancellation
+# 激活在 job 中止后仍然保留的外部 checkpoint
 env.get_checkpoint_config().enable_externalized_checkpoints(ExternalizedCheckpointCleanup.RETAIN_ON_CANCELLATION)
 
-# allow job recovery fallback to checkpoint when there is a more recent savepoint
+# 允许在有更近保存点时回退到 checkpoint
```
Comment (xintongsong) [inline docs/dev/stream/state/checkpointing.zh.md]: savepoint不翻译

**Comment by Henvealf** [comment]
> Thank for addressing the comments, @Henvealf.
> I find two occurrences of 'savepoint' which is suggested not to be translated according to the translation specifications. Besides that, I have no other concerns.

Thanks! I have fixed this.

File: docs/dev/stream/state/checkpointing.zh.md [hunk 6]
```diff
@@ -25,146 +25,138 @@ under the License.
...
-Every function and operator in Flink can be **stateful** (see [working with state](state.html) for details).
-Stateful functions store data across the processing of individual elements/events, making state a critical building block for
-any type of more elaborate operation.
+Flink 中的每个方法或算子都能够是**有状态的**（阅读 [working with state](state.html) 查看详细）。
+状态化的方法在处理单个 元素/事件 的时候存储数据，让状态成为使各个类型的算子更加精细的重要部分。
+为了让状态容错，Flink 需要为状态添加**Checkpoint（检查点）**。Checkpoint 使得 Flink 能够恢复状态和在流中的位置，从而向应用提供和无故障执行时一样的语义。
 
-In order to make state fault tolerant, Flink needs to **checkpoint** the state. Checkpoints allow Flink to recover state and positions
-in the streams to give the application the same semantics as a failure-free execution.
+[Documentation on streaming fault tolerance]({{ site.baseurl }}/zh/internals/stream_checkpointing.html) 介绍了 Flink 流计算容错机制的内部技术原理。
```
[the remark this answers is not in the document]
Comment (Henvealf) [reply docs/dev/stream/state/checkpointing.zh.md]: 好

File: docs/dev/stream/state/checkpointing.zh.md [hunk 7]
```diff
@@ -25,146 +25,138 @@ under the License.
...
+// 允许在有更近保存点时回退到 checkpoint
 env.getCheckpointConfig().setPreferCheckpointForRecovery(true);
 {% endhighlight %}
 </div>
 <div data-lang="scala" markdown="1">
 {% highlight scala %}
 val env = StreamExecutionEnvironment.getExecutionEnvironment()
 
-// start a checkpoint every 1000 ms
+// 每 1000 ms 开始一个 checkpoint
```
[the remark this answers is not in the document]
Comment (Henvealf) [reply docs/dev/stream/state/checkpointing.zh.md]: 改成了 “每 1000ms 开始一次 checkpoint”

File: docs/dev/stream/state/checkpointing.zh.md [hunk 8]
```diff
@@ -25,146 +25,138 @@ under the License.
...
+// 设置模式为精确一次 (这是默认值)
 env.getCheckpointConfig.setCheckpointingMode(CheckpointingMode.EXACTLY_ONCE)
 
-// make sure 500 ms of progress happen between checkpoints
+// 确认 checkpoint 之间的时间会进行 500 ms
 env.getCheckpointConfig.setMinPauseBetweenCheckpoints(500)
 
-// checkpoints have to complete within one minute, or are discarded
+// Checkpoint 必须在一分钟内完成，否则就会被丢弃
 env.getCheckpointConfig.setCheckpointTimeout(60000)
```
[the remark this answers is not in the document]
Comment (Henvealf) [reply docs/dev/stream/state/checkpointing.zh.md]: 丢弃是什么都不剩，放弃可能是不使用了，但是还会遗留写东西。我已经改成了“放弃”，不知哪个更符合实现。

File: docs/dev/stream/state/checkpointing.zh.md [hunk 9]
```diff
@@ -25,146 +25,138 @@ under the License.
...
-    one can define how much time needs to pass between checkpoints. If this value is set for example to *5000*, the next checkpoint will be
-    started no sooner than 5 seconds after the previous checkpoint completed, regardless of the checkpoint duration and the checkpoint interval.
-    Note that this implies that the checkpoint interval will never be smaller than this parameter.
+  - *精确一次（exactly-once） 对比 至少一次（at-least-once）*：你可以选择向 `enableCheckpointing(n)` 方法中传入一个模式来选择使用两种保证等级中的哪一种。
+    对于大多数应用来说，精确一次是较好的选择。至少一次可能与某些延迟超低（始终只有几毫秒）的应用的关联较大。
+  
+  - *Checkpoint超时（checkpoint timeout）*：如果过了这个时间，还在进行中的 checkpoint 操作就会被抛弃。
+  
+  - *checkpoint 之间的最小时间（minimum time between checkpoints）*： 为了确保流应用在 checkpoint 之间有足够的进展，可以定义在 checkpoint 之间需要多久的时间。如果值设置为了 *5000*，
+    无论 checkpoint 持续时间与间隔是多久，在前一个 checkpoint 完成的五秒后才会开始下一个 checkpoint。
```
Comment (wuchong) [inline docs/dev/stream/state/checkpointing.zh.md]:
```suggestion
    无论 checkpoint 持续时间与间隔是多久，在前一个 checkpoint 完成的至少五秒后会才开始下一个 checkpoint。
```
这里 `no sooner than 5 seconds` 很重要，我觉得得翻译出来。

File: docs/dev/stream/state/checkpointing.zh.md [hunk 10]
```diff
@@ -25,146 +25,138 @@ under the License.
...
-  - *checkpoint timeout*: The time after which a checkpoint-in-progress is aborted, if it did not complete by then.
-
-  - *minimum time between checkpoints*: To make sure that the streaming application makes a certain amount of progress between checkpoints,
-    one can define how much time needs to pass between checkpoints. If this value is set for example to *5000*, the next checkpoint will be
-    started no sooner than 5 seconds after the previous checkpoint completed, regardless of the checkpoint duration and the checkpoint interval.
-    Note that this implies that the checkpoint interval will never be smaller than this parameter.
+  - *精确一次（exactly-once） 对比 至少一次（at-least-once）*：你可以选择向 `enableCheckpointing(n)` 方法中传入一个模式来选择使用两种保证等级中的哪一种。
+    对于大多数应用来说，精确一次是较好的选择。至少一次可能与某些延迟超低（始终只有几毫秒）的应用的关联较大。
+  
+  - *Checkpoint超时（checkpoint timeout）*：如果过了这个时间，还在进行中的 checkpoint 操作就会被抛弃。
```
Comment (klion26) [inline docs/dev/stream/state/checkpointing.zh.md]:
```suggestion
  - *Checkpoint 超时（checkpoint timeout）*：如果过了这个时间，还在进行中的 checkpoint 操作就会被抛弃。
```

File: docs/dev/stream/state/checkpointing.zh.md [hunk 11]
```diff
@@ -25,146 +25,138 @@ under the License.
...
 
-Flink's checkpointing mechanism interacts with durable storage for streams and state. In general, it requires:
+  - 一个能够回放一段时间内数据的持久化数据源，例如持久化消息队列（例如 Apache Kafka、RabbitMQ、 Amazon Kinesis、 Google PubSub 等）或文件系统（例如 HDFS、 S3、 GFS、 NFS、 Ceph 等）。
+  - 存放状态的持久化存储，通常为分布式文件系统（比如 HDFS、 S3、 GFS、 NFS、 Ceph 等）。
 
-  - A *persistent* (or *durable*) data source that can replay records for a certain amount of time. Examples for such sources are persistent messages queues (e.g., Apache Kafka, RabbitMQ, Amazon Kinesis, Google PubSub) or file systems (e.g., HDFS, S3, GFS, NFS, Ceph, ...).
-  - A persistent storage for state, typically a distributed filesystem (e.g., HDFS, S3, GFS, NFS, Ceph, ...)
+## 激活与配置 Checkpoint
 
+默认情况下，Checkpoint 是禁用的。通过调用 `StreamExecutionEnvironment` 的 `enableCheckpointing(n)` 来激活 Checkpoint，里面的 *n* 是进行 Checkpoint 的间隔，单位毫秒。
```
[the remark this answers is not in the document]
Comment (wuchong) [reply docs/dev/stream/state/checkpointing.zh.md]: +1 to "启用"

**Comment by wuchong** [comment]
Hi @Henvealf  do you have time to update this?

**Comment by Henvealf** [comment]
> Hi @Henvealf do you have time to update this?

Sorry for not replying for a long time, I will update it tomorrow.

File: docs/dev/stream/state/checkpointing.zh.md [hunk 12]
```diff
@@ -25,146 +25,138 @@ under the License.
...
-// set mode to exactly-once (this is the default)
+// 设置模式为精确一次 (这是默认值)
 env.getCheckpointConfig().setCheckpointingMode(CheckpointingMode.EXACTLY_ONCE);
 
-// make sure 500 ms of progress happen between checkpoints
+// 确认 checkpoint 之间的时间会进行 500 ms
 env.getCheckpointConfig().setMinPauseBetweenCheckpoints(500);
 
-// checkpoints have to complete within one minute, or are discarded
+// Checkpoint 必须在一分钟内完成，否则就会被放弃
```
[the remark this answers is not in the document]
Comment (Henvealf) [reply docs/dev/stream/state/checkpointing.zh.md]: 所有"弃" 都统一改成了 “抛弃”。

File: docs/dev/stream/state/checkpointing.zh.md [hunk 13]
```diff
@@ -173,30 +165,26 @@ Some more parameters and/or defaults may be set via `conf/flink-conf.yaml` (see
...
 
 {% top %}
 
+## 重启策略
 
-## Restart Strategies
-
-Flink supports different restart strategies which control how the jobs are restarted in case of a failure. For more 
-information, see [Restart Strategies]({{ site.baseurl }}/dev/restart_strategies.html).
+Flink 支持不同的重启策略，来控制 job 万一故障时该如何重启。更多信息请阅读 [Restart Strategies]({{ site.baseurl }}/zh/dev/restart_strategies.html)。
```
[the remark this answers is not in the document]
Comment (Henvealf) [reply docs/dev/stream/state/checkpointing.zh.md]: 看了下会跳转。

File: docs/dev/stream/state/checkpointing.zh.md [hunk 14]
```diff
@@ -173,30 +165,26 @@ Some more parameters and/or defaults may be set via `conf/flink-conf.yaml` (see
 
 ## Selecting a State Backend
```
[the remark this answers is not in the document]
Comment (Henvealf) [reply docs/dev/stream/state/checkpointing.zh.md]: OK

**Pull request closed without merging**